game/src/common/speed.rs

```diff
@@ -149,19 +151,12 @@ impl SpeedControls {
         self.slow_down_btn.event(ctx);
         self.speed_up_btn.event(ctx);
 
-        let desired_speed = self.desired_speed();
-        if self.speed_up_btn.clicked() && desired_speed != self.speed_cap {
-            self.slider.set_percent(
-                ctx,
-                (self.slider.get_percent() + ADJUST_SPEED_PERCENT).min(1.0),
-            );
-        } else if self.slow_down_btn.clicked() && desired_speed != 0.0 {
-            self.slider.set_percent(
-                ctx,
-                (self.slider.get_percent() - ADJUST_SPEED_PERCENT).max(0.0),
-            );
+        if self.speed_up_btn.clicked() && self.speed_actual != self.speed_cap {
+            self.speed_actual += self.speed_cap * ADJUST_SPEED_PERCENT;
+        } else if self.slow_down_btn.clicked() && self.speed_actual != 0.0 {
+            self.speed_actual -= self.speed_cap * ADJUST_SPEED_PERCENT;
```

Comment:
self.speed_cap is the highest speed setting that should be possible for people to set -- 600x normally. I'm not sure multiplying by it makes sense. Also, it's possible to request negative speed here -- that's what the .max(0.0) was guarding against previously

Reply:
top speed_cap * ADJUST_SPEED_PERCENT gives the speed change.  PERCENT is 1% (0.01), so the total change is 6. Can certainly change it to less, like 0.002, so give the increment of about 1. Have you tried the code? Works well actually.

Reply:
Clicking the slow down button a few times, I can request negative speeds: https://imgur.com/HJ6711I